lib/charms/vault_k8s/v0/vault_client.py

```diff
@@ -82,80 +160,100 @@ def is_active(self) -> bool:
-            return False
-
-    def unseal(self, unseal_keys: List[str]) -> None:
-        """Unseal Vault."""
-        for unseal_key in unseal_keys:
-            self._client.sys.submit_unseal_key(unseal_key)
-        logger.info("Vault is unsealed")
-
-    def set_token(self, token: str) -> None:
-        """Set the Vault token for authentication."""
-        self._client.token = token
+    def enable_audit_device(self, device_type: AuditDeviceType, path: str) -> None:
+        """Enable a new audit device at the supplied path.
 
-    def approle_login(self, role_id: str, secret_id: str) -> None:
-        """Authenticate to Vault using AppRole."""
-        self._client.auth.approle.login(role_id=role_id, secret_id=secret_id, use_token=True)
+        Args:
+            device_type: One of three available device types
+            path: The path that will receive audit logs
+        """
+        try:
+            self._client.sys.enable_audit_device(
+                device_type=device_type.value,
+                options={"file_path": path},
+            )
+            logger.info("Enabled audit device %s for path %s", device_type.value, path)
+        except InvalidRequest as e:
+            errors = e.json.get("errors")
+            if len(errors) == 1 and errors[0].startswith("path is already in use"):
+                logger.info("Audit device already enabled.")
+            else:
+                raise e
+
+    def enable_approle_auth_method(self) -> None:
+        """Enable approle auth method."""
+        try:
+            self._client.sys.enable_auth_method("approle")
+            logger.info("Enabled approle auth method.")
+        except InvalidRequest as e:
+            errors = e.json.get("errors")
+            if len(errors) == 1 and errors[0].startswith("path is already in use"):
+                logger.info("Approle already enabled.")
+            else:
+                raise e
 
-    def configure_charm_access_policy(self, name: str) -> None:
-        """Create/update a policy within vault for the charm to access vault."""
-        with open("src/templates/charm_policy.hcl", "r") as f:
-            charm_policy = f.read()
-        self._client.sys.create_or_update_policy(name=name, policy=charm_policy)
-        logger.debug(f"Created charm policy: {name}")
+    def configure_policy(
+        self, policy_name: str, policy_path: str, mount: Optional[str] = None
+    ) -> None:
+        """Create/update a policy within vault.
 
-    def remove_raft_node(self, node_id: str) -> None:
-        """Remove raft peer."""
-        self._client.sys.remove_raft_node(server_id=node_id)
-        logger.info("Removed raft node %s", node_id)
+        Args:
+            policy_name: Name of the policy to create
+            policy_path: The path of the file where the policy is defined, ending with .hcl
+            mount: Formats the policy string with the given mount path to create dynamic policies
```

Comment:
Bonus points for not including the type here when it's already type hinted :laughing: